pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -87,8 +88,9 @@ def qfunc(x, y, z):
      1: ──╭C──────RY(2)──RY(-2)──│───────┤
      2: ──╰X──────H──────────────╰C──────┤
     """
-    # Make a working copy of the list to traverse
-    list_copy = tape.operations.copy()
+    # Expand away adjoint ops
+    expanded_tape = tape.expand(stop_at=lambda obj: not isinstance(obj, Adjoint))
+    list_copy = expanded_tape.operations
```

Comment:
Can you explain this change? How come the expansion with a stopping criterion will cancel adjoint ops (as the comment implies)?

Reply:
If it's not an `Adjoint`, then it will not be expanded. It will be "stopped at".

If it is an `Adjoint`, it will be expanded. `"Adjoint(RX)"` will be decomposed to `RX`, which can then be combined with a different `RX`.

Reply:
Oh ok!